cloudinit/analyze/dump.py

```diff
@@ -44,14 +44,21 @@ def parse_timestamp(timestampstr):
         dt = datetime.strptime(timestampstr, CLOUD_INIT_ASCTIME_FMT)
         timestamp = dt.strftime("%s.%f")
     else:
-        # allow date(1) to handle other formats we don't expect
+        # allow GNU date(1) to handle other formats we don't expect
+        # This may throw a ValueError if no GNU date can be found
         timestamp = parse_timestamp_from_date(timestampstr)
 
     return float(timestamp)
 
 
 def parse_timestamp_from_date(timestampstr):
-    out, _ = subp.subp(["date", "+%s.%3N", "-d", timestampstr])
+    cmd = "date"
+    if not util.is_Linux():
+        if subp.which("gdate"):
+            cmd = "gdate"
+        else:
+            raise ValueError("Unable to parse timestamps without GNU date(1)")
```

Comment:
I think we should include the concrete timestamp here, so we can learn and adapt

Reply:
By that do you mean something like this?
```
           raise ValueError(f"Unable to parse timestamps without GNU date: [{timestampstr}]")
```

I think it would certainly help debugging to get this information.

Reply:
yes. that sort of thing.
wanna put that into a "suggestion", so i can merge it from my phone?
(I'm off the laptop)

Reply:
```suggestion
            raise ValueError(
                f"Unable to parse timestamps without GNU date: [{timestampstr}]"
            )
```

Reply:
> yes. that sort of thing. wanna put that into a "suggestion", so i can merge it from my phone? (I'm off the laptop)

sure, just did